openfecwebapp/data_mappings.py

```diff
@@ -0,0 +1,195 @@
+    c['url'] = '/committees/' + ac.get('committee_id', '')
+    return c
+
+# maps totals to be shown on candidate and committee pages
+def map_totals(t):
+    reports = t['results'][0]['reports'][0]
+    totals = t['results'][0]['totals'][0]
+    totals_mapped = {}
+    value_map = {
+        'total_receipts': totals.get('receipts'),
+        'total_disbursements': totals.get('total_disbursements'),
+        'total_cash': reports.get('cash_on_hand_end_period'),
+        'total_debt': reports.get('debts_owed_by_committee')
+    }
+
+    # format totals data in US dollars
+    for v in value_map:
+        if value_map[v]:
+            totals_mapped[v] = locale.currency(
+            value_map[v], grouping=True)
+        else:
+            totals_mapped[v] = 'unavailable'
+
+    if reports.get('report_year'):
+        totals_mapped['report_year'] = str(int(reports['report_year']))
+
+    # "calculated from" on site
+    if reports.get('election_cycle'):
+        cycle_minus_one = str(int(reports['election_cycle']) - 1)
+        totals_mapped['years_totals'] = cycle_minus_one 
+        totals_mapped['years_totals'] += ' - ' 
+        totals_mapped['years_totals'] += str(reports['election_cycle'])
+
+    # "source:" on site
+    if reports.get('report_type_full'):
+        totals_mapped['report_desc'] = re.sub('{.+}', 
+            '', reports['report_type_full']) 
+
+    return totals_mapped
+
+# we want to show the committees on their related candidate 
+# pages in this order, with primary committees on top
+committee_type_map = {
+    'A': 'authorized_committees',
+    'D': 'leadership_committees',
+    'J': 'joint_committees'
+}
+
+def map_candidate_page_values(c):
+    candidate = map_candidate_table_values(c)
+
+    if c.get('elections'):
+        c_e = c['elections'][0]
+        candidate['incumbent_challenge'] = c_e.get(
+            'incumbent_challenge_full', '')
+        if c_e.get('primary_committee'):
+            candidate['primary_committee'] = _map_committee_values(
+                c_e['primary_committee'])
+            candidate['related_committees'] = True
+
+        # affiliated committees = committee_type_map, 
+        # plus more we ignore for the candidate pages
+        if c_e.get('affiliated_committees'):
+            candidate['affiliated_committees'] = []
+            for i in range(len(c_e['affiliated_committees'])):
+                candidate['affiliated_committees'].append(
+                    _map_committee_values(c_e[
+                        'affiliated_committees'][i]))
+
+            candidate['authorized_committees'] = []
+            candidate['leadership_committees'] = []
+            candidate['joint_committees'] = []
+
+            for i in range(len(candidate['affiliated_committees'])):
+                cmte = candidate['affiliated_committees'][i]
+                cmte_type = cmte['designation_code']
+                # drop anything that's not of the types we're
+                # interested in
```

Comment:
I wonder if this sort of filter should eventually make its way into the API, rather than forcing the client to do it.